Source/Core/Common/Assembler/GekkoLexer.cpp

```diff
@@ -182,6 +183,10 @@ std::optional<T> EvalIntegral(TokenType tp, std::string_view val)
     return std::accumulate(val.begin() + 2, val.end(), T{0}, bin_step);
   case TokenType::GPR:
   case TokenType::FPR:
+    if (CaseInsensitiveEquals(val, "sp"))
+      return T{1};
+    if (CaseInsensitiveEquals(val, "rtoc"))
+      return T{2};
```

Comment:
I just realized this can be improved a little further.
```suggestion
  case TokenType::GPR:
    if (CaseInsensitiveEquals(val, "sp"))
      return T{1};
    if (CaseInsensitiveEquals(val, "rtoc"))
      return T{2};
    [[fallthrough]];
  case TokenType::FPR:
```
This way, FPRs won't be checking if they are "sp" or "rtoc" when they logically never should be.